tests/drivers/ipm/src/main.c

```diff
@@ -65,9 +67,9 @@ static struct ipm_console_receiver_config_info receiver_config = {
 };
 
 struct ipm_console_receiver_runtime_data receiver_data;
-DEVICE_INIT(ipm_console_recv0, "ipm_recv0", ipm_console_receiver_init,
+DEVICE_AND_API_INIT(ipm_console_recv0, "ipm_recv0", ipm_console_receiver_init,
 	    &receiver_data, &receiver_config,
-	    APPLICATION, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);
+	    APPLICATION, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT, NULL);
```

Comment:
Same.  Why even bother having DEVICE_INIT() if you can't use it as intended?

Reply:
did you mean DEVICE_AND_API_INIT()  instead of DEVICE_INIT() in your comment? Can you pls elaborate?
I have modified DEVICE_INIT() to DEVICE_AND_API_INIT() as DEVICE_INIT() will tagged as deprecated, no driver_api is being used here, so assigned NULL